aws/resource_aws_iot_thing_test.go

```diff
@@ -0,0 +1,159 @@
+					resource.TestCheckResourceAttr("aws_iot_thing.test", "attributes.Two", "TwoTwo"),
+					resource.TestCheckResourceAttr("aws_iot_thing.test", "attributes.Answer", "42"),
+					resource.TestCheckResourceAttrSet("aws_iot_thing.test", "arn"),
+					resource.TestCheckResourceAttrSet("aws_iot_thing.test", "default_client_id"),
+					resource.TestCheckResourceAttrSet("aws_iot_thing.test", "version"),
+				),
+			},
+			{ // Update attribute
+				Config: testAccAWSIotThingConfig_full(thingName, typeName, "differentOne"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("aws_iot_thing.test", "name", thingName),
+					resource.TestCheckResourceAttr("aws_iot_thing.test", "thing_type_name", typeName),
+					resource.TestCheckResourceAttr("aws_iot_thing.test", "attributes.%", "3"),
+					resource.TestCheckResourceAttr("aws_iot_thing.test", "attributes.One", "11111"),
+					resource.TestCheckResourceAttr("aws_iot_thing.test", "attributes.Two", "TwoTwo"),
+					resource.TestCheckResourceAttr("aws_iot_thing.test", "attributes.Answer", "differentOne"),
+					resource.TestCheckResourceAttrSet("aws_iot_thing.test", "arn"),
+					resource.TestCheckResourceAttrSet("aws_iot_thing.test", "default_client_id"),
+					resource.TestCheckResourceAttrSet("aws_iot_thing.test", "version"),
+				),
+			},
+			{ // Remove thing type association
+				Config: testAccAWSIotThingConfig_basic(thingName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("aws_iot_thing.test", "name", thingName),
+					resource.TestCheckResourceAttr("aws_iot_thing.test", "attributes.%", "0"),
+					resource.TestCheckResourceAttr("aws_iot_thing.test", "thing_type_name", ""),
+					resource.TestCheckResourceAttrSet("aws_iot_thing.test", "arn"),
+					resource.TestCheckResourceAttrSet("aws_iot_thing.test", "default_client_id"),
+					resource.TestCheckResourceAttrSet("aws_iot_thing.test", "version"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSIotThing_importBasic(t *testing.T) {
+	resourceName := "aws_iot_thing.test"
+	rString := acctest.RandString(8)
+	thingName := fmt.Sprintf("tf_acc_thing_%s", rString)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSIotThingTypeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSIotThingConfig_basic(thingName),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSIotThingDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).iotconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_iot_thing" {
+			continue
+		}
+
+		params := &iot.DescribeThingInput{
+			ThingName: aws.String(rs.Primary.ID),
+		}
+
+		_, err := conn.DescribeThing(params)
+		if err == nil {
```

Comment:
Should we flip this around to return `nil` on `iot.ErrCodeResourceNotFoundException` error and return the actual error on others (e.g. permission errors)?